File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        last_unique = float('inf')
        k = 0
        for r in range(len(nums)):
            if nums[r] != last_unique:
                last_unique = nums[r]
                nums[r], nums[l] = nums[l], nums[r]
                k += 1
                l += 1
        return k
                

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const removeDuplicates = (nums) => {
  let left = 0

  for (let right = 1; right < nums.length; right++) {
    if (nums[right] !== nums[left]) {
      left++
      nums[left] = nums[right]
    }
  }
  return left + 1
}